weave/tests/test_weave_client.py

```diff
@@ -480,6 +480,26 @@ def add3(x, y, z):
     assert obj2["b"].name == "op-add3"
 
 
+def test_op_mismatch_project_ref(client):
```

Comment:
please also add some more tests:
1. Publish an object to another project, get it, then re-publish it to a new project
2. Publish an object to another project, get it, then add it to a nested property of the a new object entirely, then publish, ensuring all refs are correct
3. Publish an op and object to a project, then use them with a new project - verify all correctness.